src/node.cc

```diff
@@ -1110,6 +1111,57 @@ bool ShouldAbortOnUncaughtException(Isolate* isolate) {
 }
 
 
+void PromiseHook(PromiseHookType type,
+                 Local<Promise> promise,
+                 Local<Value> parent) {
+  Environment* env = Environment::GetCurrent(Isolate::GetCurrent());
+  Local<Context> context = env->context();
+
+  if (type == PromiseHookType::kInit && env->in_domain()) {
```

Comment:
Again, I don't have much context, but in_domain() check seems important, are you sure you don't need to do it for the other cases?